src/main/java/com/tabnine/inline/TabnineDocumentListener.java

```diff
@@ -59,7 +61,7 @@ public void documentChanged(@NotNull DocumentEvent event) {
   private boolean shouldIgnoreChange(DocumentEvent event, Editor editor, int offset) {
     Document document = event.getDocument();
 
-    if (SuggestionsMode.getSuggestionMode() != SuggestionsMode.INLINE || event.getNewLength() < 1) {
+    if (event.getNewLength() < 1 || !suggestionsModeService.getSuggestionMode().isInlineEnabled()) {
```

Comment:
switched conditions here because for some reason the tests in `PredictionBehaviourIntegrationTests.java` yielded a doc change with empty string (they shouldn't 🤷🏻), and `suggestionsModeService.getSuggestionMode()` returned `null` 😢 
Looks like my kombina with this service helped the other tests and fucked up these tests... I really can't figure out why, both the hybrid tests and the tests in `PredictionBehaviourIntegrationTests` are extending the same abstract class with the same mocks and flow, I can't figure out what's different :(

Well, as long as it works...
@dimaabramovich @yairco1990

Reply:
Found the issue, the difference was that the tests in `PredictionBehaviourIntegrationTests` didn't register the immediate application:
```java
    ApplicationManager.setApplication(mockedApplicationWhichInvokesImmediately(), this);
```
I moved this logic to the abstract class since there's no reason not to use it in all these tests 🙂